pkg/processor/runtime/python/py/_nuclio_async_wrapper.py

```diff
@@ -0,0 +1,308 @@
+# Copyright 2023 The Nuclio Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import traceback
+import time
+import json
+import nuclio_sdk
+import nuclio_sdk.helpers
+import nuclio_sdk.json_encoder
+import nuclio_sdk.logger
+import socket
+import asyncio
+import selectors
```

Comment:
Is there a more native alternative to the combination of asyncio and selectors? perhaps only asyncio? I believe you can achieve it using asyncio / socket only. am I wrong?

Reply:
@liranbg after trying it, I'm not sure about asyncio raw approach as it will make as to go to absolutely different code with processing from socket. since asyncio approach suggests to use their `reader: asyncio.StreamReader, writer: asyncio.StreamWriter` as a reader and writer which means that we can't use our existing code for socket communication. I tried to get a socket from reader and use it however 1-it's not a recommended approach 2 - I encountered issues with file descriptors. So not sure we want to put some effort here right now. We can create a research task for future as this is not something we can answer on fast.